simbad/mr/phaser_mr.py

```diff
@@ -214,44 +251,47 @@ def run(self):
         pdbin = os.path.join(self.work_dir, os.path.basename(self.pdbin))
         shutil.copyfile(self.pdbin, pdbin)
 
+        i = InputMR_DAT()
+        i.setHKLI(hklin)
+
+        if self.hires:
+            i.setHIRES(self.hires)
+        if self.autohigh:
+            i.setRESO_AUTO_HIGH(self.autohigh)
+        if self.i != "None" and self.sigi != "None":
+            i.setLABI_I_SIGI(self.i, self.sigi)
+        elif self.f != "None" and self.sigf != "None":
+            i.setLABI_F_SIGF(self.f, self.sigf)
+        else:
+            msg = "No flags for intensities or amplitudes have been provided"
+            raise RuntimeError(msg)
         if self.sgalternative == "all":
-            sgalternative = "ALL"
+            i.setSGAL_SELE("ALL")
         elif self.sgalternative == "enant":
-            sgalternative = "HAND"
+            i.setSGAL_SELE("HAND")
         else:
-            sgalternative = "NONE"
-
-        if self.timeout == 0:
-            kill_command = ""
-        else:
-            kill_command = "KILL TIME {0}".format(self.timeout)
-
-        key = """#---PHASER COMMAND SCRIPT GENERATED BY SIMBAD---
-        MODE MR_AUTO
-        ROOT "phaser_mr_output"'
-        {0}
-        #---DEFINE DATA---
-        HKLIN {1}
-        LABIN I={2} SIGI={3}
-        SGALTERNATIVE SELECT {4}
-        #---DEFINE ENSEMBLES---
-        ENSEMBLE ensemble1 &
-            PDB "{5}" RMS 0.6
-        #---DEFINE COMPOSITION---
-        COMPOSITION BY SOLVENT
-        COMPOSITION PERCENTAGE {6}
-        #---SEARCH PARAMETERS---
-        SEARCH ENSEMBLE ensemble1 NUMBER {7}"""
-        key = key.format(kill_command, hklin, self.i, self.sigi, sgalternative, pdbin, self.solvent, self.nmol)
-
-        Phaser.phaser(self.logfile, key)
-
-        # Move the output hkl file to specified filename
-        if os.path.isfile(os.path.join(self.work_dir, 'phaser_mr_output.1.mtz')):
-            shutil.move(os.path.join(self.work_dir, 'phaser_mr_output.1.mtz'), self.hklout)
-        # Move output pdb file to specified filename
-        if os.path.isfile(os.path.join(self.work_dir, 'phaser_mr_output.1.pdb')):
-            shutil.move(os.path.join(self.work_dir, 'phaser_mr_output.1.pdb'), self.pdbout)
+            i.setSGAL_SELE("NONE")
+        i.setMUTE(True)
+        r = runMR_DAT(i)
+
+        if r.Success():
+            i = InputMR_AUTO()
+            i.setJOBS(1)
+            i.setREFL_DATA(r.getREFL_DATA())
+            i.setROOT("phaser_mr_output")
+            i.addENSE_PDB_RMS("PDB", pdbin, 0.6)
```

Comment:
Maybe this rms value should be made a class attribute in case somebody wants to define it.

Reply:
In the next few weeks i'm planning to explore this number more with the use of ensembles so it might not be worth changing yet

Reply:
I'm not sure it's a matter of changing the value but probably worth exposing now ;-)